File: upande_payroll/upande_payroll/report/affordable_housing_levy/affordable_housing_levy.js
// Copyright (c) 2026, Teresia and contributors
// For license information, please see license.txt

frappe.query_reports["Affordable Housing Levy"] = {
	filters: [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: frappe.defaults.get_user_default("Company"),
			reqd: 1,
		},
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
			default: frappe.datetime.month_start(),
			reqd: 1,
		},
		{
			fieldname: "to_date",
			label: __("To Date"),
			fieldtype: "Date",
			default: frappe.datetime.month_end(),
			reqd: 1,
		},
		{
			fieldname: "docstatus",
			label: __("Payslip Status"),
			fieldtype: "Select",
			options: "Submitted\nDraft\nCancelled",
			default: "Submitted",
		},
		{
			fieldname: "department",
			label: __("Department"),
			fieldtype: "Link",
			options: "Department",
		},
		{
			fieldname: "employee",
			label: __("Employee"),
			fieldtype: "Link",
			options: "Employee",
			get_query() {
				return {
					filters: { company: frappe.query_report.get_filter_value("company") },
				};
			},
		},
	],
};

File: upande_payroll/upande_payroll/report/affordable_housing_levy/affordable_housing_levy.py
# Copyright (c) 2026, Teresia and contributors
# For license information, please see license.txt

from frappe import _

from upande_payroll.statutory_reports import build

SPEC = {
	"title": _("Affordable Housing Levy"),
	"employee_keys": ("housing_levy_employee",),
	"employer_keys": ("housing_levy_employer",),
	"columns": [
		("employee_number", _("Payroll No"), "Data", 120),
		("national_id", _("ID Number"), "Data", 150),
		("employee_name", _("Employee Name"), "Data", 250),
		("kra_pin", _("KRA PIN"), "Data", 150),
		("gross_salary", _("Gross Salary"), "Currency", 180),
		("basic_salary", _("Basic Salary"), "Currency", 150),
		("member_contribution", _("Member Contribution"), "Currency", 170),
		("employer_contribution", _("Employer Contribution"), "Currency", 170),
		("total_contribution", _("Total Contribution"), "Currency", 160),
	],
}


def execute(filters=None):
	return build(filters, SPEC)
